Utilities/HttpServer.cs

```diff
@@ -98,6 +101,58 @@ public class HttpServer {
       }
     }
 
+    public static IDictionary<string, string> ToDictionary(NameValueCollection col) {
+      IDictionary<string, string> dict = new Dictionary<string, string>();
+      foreach (var k in col.AllKeys) {
+        dict.Add(k, col[k]);
+      }
+      return dict;
+    }
+
+    public SensorNode findSensor(Node node, string id) {
+      if (node is SensorNode) {
+        SensorNode sNode = (SensorNode)node;
+        if (sNode.Sensor.Identifier.ToString() == id)
+          return sNode;
+      }
+      foreach (Node child in node.Nodes) {
+        SensorNode s = findSensor(child, id);
+        if (s != null) {
+          return s;
+        }
+      }
+      return null;
+    }
+
+    public bool doSetValue(string id, string value) {

```

Comment:
Maybe something like 'SetSensorControlValue'?

Reply:
Renamed to SetSensorControlValue.